pypeit/images/sciencecube.py

```diff
@@ -0,0 +1,256 @@
+        """
+        Updates the CR mask values in self.mask
+        through a call to ImageMask.build_crmask which
+        generates a new CR mask and then a call to
+        ImageMask.update_mask_cr() which updates self.mask
+
+        Args:
+            subtract_img (np.ndarray, optional):
+                If provided, this is subtracted from self.image prior to
+                CR masking
+        """
+        # Generate the CR mask (and save in self.crmask)
+        super(ScienceImage, self).build_crmask(self.spectrograph, self.det,
+                                               self.par, self.image,
+                                               utils.inverse(self.ivar),
+                                               subtract_img=subtract_img).copy()
+        # Now update the mask
+        super(ScienceImage, self).update_mask_cr(self.crmask)
+
+    def __sub__(self, other):
+        """
+        Subtract a ScienceImage object from another
+        Extras (e.g. ivar, masks) are included if they are present
+
+        Args:
+            other (ScienceImage):
+
+        Returns:
+            ScienceImage:
+
+        """
+        if not isinstance(other, ScienceImage):
+            msgs.error("Misuse of the subtract method")
+        # Images
+        newimg = self.image - other.image
+
+        # Mask time
+        outmask_comb = (self.mask == 0) & (other.mask == 0)
+
+        # Variance
+        if self.ivar is not None:
+            new_ivar = utils.inverse(utils.inverse(self.ivar) + utils.inverse(other.ivar))
+            new_ivar[np.invert(outmask_comb)] = 0
+        else:
+            new_ivar = None
+
+        # RN2
+        if self.rn2img is not None:
+            new_rn2 = self.rn2img + other.rn2img
+        else:
+            new_rn2 = None
+
+        # Files
+        new_files = self.files + other.files
+
+        # Instantiate
+        new_sciImg = ScienceImage(self.spectrograph, self.det, self.par,
+            newimg, new_ivar, self.bpm, rn2img=new_rn2, files=new_files)
+        #TODO: KW properly handle adding the bits
+        crmask_diff = new_sciImg.build_crmask()
+        # crmask_eff assumes evertything masked in the outmask_comb is a CR in the individual images
+        new_sciImg.crmask = crmask_diff | np.invert(outmask_comb)
+        # Note that the following uses the saturation and mincounts held in
+        # self.spectrograph.detector[self.det-1]
+        new_sciImg.build_mask()
+
+        return new_sciImg
+
+    def __repr__(self):
+        repr = '<{:s}: files={}'.format(self.__class__.__name__, self.files)
+        # Image
+        rdict = {}
+        for attr in ['image', 'ivar', 'rn2img', 'crmask', 'mask']:
+            if getattr(self, attr) is not None:
+                rdict[attr] = True
+            else:
+                rdict[attr] = False
+        repr += ' images={}'.format(rdict)
+        repr = repr + '>'
+        return repr
+
+
+def build_from_file_list(spectrograph, det, par, bpm,
```

Comment:
I think this doesn't belong..

Maybe this whole module should be removed?

Reply:
If so, also the lines in `buildimages.py`

Reply:
OK - removed (it seems like nothing is calling this code)!